src/qnngds/utilities.py

```diff
@@ -193,6 +198,88 @@ def calculate_contact_w(
         max_circuit_port_width = max([port.width for port in circuit_ports])
         return max(max_circuit_port_width, self.contact_l)
 
+class ConnectionToPCB():
```

Comment:
all of these classes should have some sort of docstring, and their methods should as well